client/web/compose/src/components/Public/Record/Modal.vue

```diff
@@ -0,0 +1,132 @@
+<template>
+  <b-modal
+    id="record-modal"
+    v-model="showRecordModal"
+    scrollable
+    body-class="p-0"
+    footer-class="p-0"
+    size="xl"
+  >
+    <template #modal-title>
+      <portal-target
+        name="record-modal-header"
+      />
+    </template>
+
+    <view-record
+      :namespace="namespace"
+      :page="page"
+      :module="module"
+      :record-i-d="recordID"
+      :show-record-modal="showRecordModal"
+    />
+
+    <template #modal-footer>
+      <portal-target
+        name="record-modal-footer"
+        class="w-100 m-0"
+      />
+    </template>
+  </b-modal>
+</template>
+
+<script>
+import record from 'corteza-webapp-compose/src/mixins/record'
+import { compose } from '@cortezaproject/corteza-js'
+import ViewRecord from 'corteza-webapp-compose/src/views/Public/Pages/Records/View'
+import { mapGetters } from 'vuex'
+
+export default {
+  i18nOptions: {
+    namespaces: 'block',
+  },
+
+  name: 'RecordModal',
+
+  components: {
+    ViewRecord,
+  },
+
+  mixins: [
+    record,
+  ],
+
+  props: {
+    namespace: {
+      type: compose.Namespace,
+      required: true,
+    },
+  },
+
+  data () {
+    return {
+      showRecordModal: false,
+      recordID: null,
+      module: null,
+      page: null,
+    }
+  },
+
+  computed: {
+    ...mapGetters({
+      getModuleByID: 'module/getByID',
+      pages: 'page/set',
+    }),
+  },
+
+  watch: {
+    showRecordModal: function(){
+      if(!this.showRecordModal){
+        this.$router.replace({ query: { } })
```

Comment:
Why this? You need to replace it when the modal is closed, and you can do this in a better way rather than the watcher